### PR TITLE
CopyFiles keep permissions

### LINKED ISSUE #534: Copying files should keep permissions
Files moved via [CopyFiles()](https://github.com/suse-edge/edge-image-builder/blob/main/pkg/fileio/file_io.go#L110) will always default to setting 0644 perms. Investigate all usages, confirm that this default behaviour is not desired and adjust the code to use the input perms instead.

### PR DESCRIPTION
**Pull request by dbw7**
Closes #534

### CODE REVIEW

File: pkg/fileio/file_io.go [hunk 1]
```diff
@@ -107,7 +107,17 @@ func CopyFiles(src, dest, ext string, copySubDir bool) error {
 				continue
 			}
 
-			err := CopyFile(sourcePath, destPath, NonExecutablePerms)
+			perms := NonExecutablePerms
+			if maintainPerms {
```
Comment (atanasdinov) [inline pkg/fileio/file_io.go:111]: I'm not quite convinced - why do we need this param?

Comment (dbw7) [reply pkg/fileio/file_io.go:111]: So my idea here is that for things that we use internally, we don't need to maintain the original permissions, but for things that the user provides to be on the final image that they manage and we don’t, it should be the original permissions they specified. To this, I thought having that parameter would be a clean solution to allow us to easily specify how we want to use the function wherever we need to. The usage itself though is up for discussion.

Comment (atanasdinov) [reply pkg/fileio/file_io.go:111]: I still think this is unnecessary complexity - none of the files that we transfer are executable in the first place.

Comment (agracey) [reply pkg/fileio/file_io.go:111]: Isn't this for the custom files that a user can place in a tree?

Comment (jdob) [reply pkg/fileio/file_io.go:111]: Another consideration here is that we'll eventually need to maintain SELinux contexts. In that light, I think the approach of "always maintain whatever is on the filesystem" is probably a better approach. If I try to step back as a completely new user, I think that's what I'd expect the behavior to be.

**Review by jdob — CHANGES_REQUESTED** [review]
I think I'm with Atanas here. Let's maintain permissions in every instance. If we find an exception case in the future, we can add it back. Even then, I think the behavior should be more of "maintain permissions by default unless explicit permissions are specified".

File: pkg/fileio/file_io.go [hunk 2]
```diff
@@ -106,12 +106,20 @@ func CopyFiles(src, dest, ext string, copySubDir bool) error {
...
 			}
-			info, infoErr := file.Info()
-			if infoErr != nil {
-				return fmt.Errorf("reading file info %w", infoErr)
+
+			var perms os.FileMode
+			if overridePerms != nil {
+				perms = *overridePerms
+			} else {
+				fileInfo, InfoErr := file.Info()
```
Comment (atanasdinov) [inline pkg/fileio/file_io.go:114]: Can we revert the upper case for `InfoErr`?

**Review by atanasdinov — APPROVED** [review]

**Review by atanasdinov — APPROVED** [review]

**Review by jdob — APPROVED** [review]

**Pull request merged**